docs/tutorial/process-model.md

```diff
@@ -0,0 +1,203 @@
+> web content.
+
+Because the `BrowserWindow` module is an [`EventEmitter`][event-emitter], you can also
+add handlers for various user events (for example, minimizing or maximizing your window).
+
+```js title='main.js'
+const { BrowserWindow } = require('electron')
+
+const win = new BrowserWindow({ width: 800, height: 1500 })
+win.loadURL('https://github.com')
+
+const contents = win.webContents
+console.log(contents)
+```
+
+When a `BrowserWindow` instance is destroyed, its corresponding renderer process gets
+terminated as well.
+
+[browser-window]: ../api/browser-window.md
+[web-embed]: ./web-embeds.md
+[web-contents]: ../api/web-contents.md
+[event-emitter]: https://nodejs.org/api/events.html#events_class_eventemitter
+
+### Application lifecycle
+
+The main process also controls your application's lifecycle through Electron's
+[`app`][app] module. This module provides a large set of events and methods
+that you can use to add custom application behaviour (for instance, programatically
+quitting your application, modifying the application dock, or showing an About panel).
+
+As a practical example, the app shown in the [quick start guide][quick-start-lifecycle]
+uses `app` APIs to create a more native application window experience.
+
+```js title='main.js'
+// quitting the app when no windows are open on macOS
+app.on('window-all-closed', function () {
+  if (process.platform !== 'darwin') app.quit()
+})
+```
+
+[app]: ../api/app.md
+[quick-start-lifecycle]: ./quick-start.md#manage-your-windows-lifecycle
+
+### Native APIs
+
+To extend Electron's features beyond being a Chromium wrapper for web contents, the
+main process also adds custom APIs to. Electron exposes various modules that control
+native desktop functionality, such as menus, dialogs, and tray icons.
+
+For a full list of Electron's main process modules, check out our API documentation.
+
+## The renderer process
+
+Each Electron app spawns a separate renderer process for each open `BrowserWindow`
+(and each web embed). As its name implies, a renderer is responsible for
+*rendering* web content. For all intents and purposes, code ran in renderer processes
+should behave according to web standards (insofar as Chromium does, at least).
+
+Therefore, all user interfaces and app functionality within a single browser
+window should be written with the same tools and paradigms that you use on the
+web.
+
+Although explaining every web spec is out of scope for this guide, the bare minimum
+to understand is:
+
+* An HTML file is your entry point for the renderer process.
+* UI styling is added through Cascading Style Sheets (CSS).
+* Executable JavaScript code can be added through `<script>` elements.
+
+Moreover, this also means that the renderer has no direct access to `require`
+or other Node.js APIs. In order to import NPM modules in the renderer, you must use
+the same bundler toolchains (for example, `webpack` or `parcel`) that you use for web
+apps.
```

Comment:
...or use `<script src="relative/path/to/node_modules/modulename/dist/index.js"></script>`, but that can get tedious.

Reply:
I feel like I can omit that detail 🤔